tap_logmeinrescue/streams/base.py

```diff
@@ -191,22 +237,26 @@ def sync_data(self, parent_ids):
 
                     ctr.increment(amount=len(parsed_response['rows']))
 
+                # technician_id acts as a substream bookmark so that we
+                # can pick up in a single stream where we left off.
                 self.state = incorporate(
                     self.state, table, 'technician_id', parent_id)
-                self.state = incorporate(
-                    self.state, table, 'start_date', start_date)
-                self.state = save_state(self.state)
+                # There's no need to save `start_date` here. Even in the
+                # case of the first run the config start_date won't change
+                # so we're safe. It's acceptable to update start_date only
+                # after we're done the sync for this whole window.
+                save_state(self.state)
 
-            start_date = end_date
-            end_date = start_date + timedelta(days=7)
             technician_id = 0
 
             self.state = incorporate(
-                self.state, table, 'start_date', start_date)
+                self.state, table, 'start_date', end_date)
+            # Because we go through all the technicians every time we sync
+            # we need to start over by resetting the technician_id sub
+            # bookmark to 0.
             self.state = incorporate(
-                self.state, table, 'technician_id', technician_id,
-                force=True)
-            self.state = save_state(self.state)
+                self.state, table, 'technician_id', 0, force=True)
+            save_state(self.state)
```

Comment:
`save_state` returns nothing so we were overwriting `self.state` with `None` every time.